src/deepEqual.js

```diff
@@ -23,8 +23,55 @@
  *
  */
 
+function isEquals(value1, value2) {
+  if (typeof value1 === typeof value2) {
+    if (value1 === value2) {
+      return true;
+    }
+    return false;
+  }
```

Comment:
all these lines can be simplified into 
return typeof value1 === typeof vallue2 && value1 == value2;